src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -89,28 +90,124 @@ bool vtkDataMeshReader::read(const QString& path) {
             return false;
         }
 
-        try
-        {
-            dataSet->Read(path.toLocal8Bit().constData());
```

Comment:
I understood why CARTO files were not loaded properly before your last commit, it's because you don't use `vtkMetaSurfaceMesh::Read` for VTK files. It was yet a good method that was adapting according to the type of mesh.
Can't you put your changes inside this method, I mean inside `vtkMetaSurfaceMesh::ReadVtkFile` and `vtkMetaSurfaceMesh::ReadVtkFile` ?